cert-manager/controllers/keptnwebhookcontroller/certificate_secret.go

```diff
@@ -0,0 +1,123 @@
+package keptnwebhookcontroller
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	"github.com/keptn/lifecycle-toolkit/cert-manager/kubeobjects"
+	"github.com/pkg/errors"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type certificateSecret struct {
+	secret          *corev1.Secret
+	certificates    *Certs
+	existsInCluster bool
+}
+
+func newCertificateSecret() *certificateSecret {
+	return &certificateSecret{}
+}
+
+func (certSecret *certificateSecret) setSecretFromReader(ctx context.Context, apiReader client.Reader, namespace string, log logr.Logger) error {
+	query := kubeobjects.NewSecretQuery(ctx, nil, apiReader, log)
+	secret, err := query.Get(types.NamespacedName{Name: buildSecretName(), Namespace: namespace})
+
+	if k8serrors.IsNotFound(err) {
+		certSecret.secret = kubeobjects.NewSecret(buildSecretName(), namespace, map[string][]byte{})
+		certSecret.existsInCluster = false
+	} else if err != nil {
+		return errors.WithStack(err)
+	} else {
+		certSecret.secret = &secret
+		certSecret.existsInCluster = true
+	}
+
+	return nil
+}
+
+func (certSecret *certificateSecret) isRecent() bool {
```

Comment:
if I understand it correctly, this function compares if certificate and secret are the same, why it is called `isRecent()` ? I would maybe expect `isEqual` or something like that...

Reply:
this compares if the certificate is not expired